rpc/rpcserver/server.go

```diff
@@ -139,6 +141,201 @@ func decodeAddress(a string, params *chaincfg.Params) (dcrutil.Address, error) {
+		MaxPriceScale:             t.ticketbuyerCfg.MaxPriceScale,
+		MaxInMempool:              t.ticketbuyerCfg.MaxInMempool,
+		PoolAddress:               req.PoolAddress,
+		PoolFees:                  req.PoolFees,
+		SpreadTicketPurchases:     t.ticketbuyerCfg.SpreadTicketPurchases,
+		TicketAddress:             req.TicketAddress,
+		TxFee:                     t.ticketbuyerCfg.TxFee,
+	}
+	err := t.loader.StartTicketPurchase(req.Passphrase, config)
+	if err == loader.ErrTicketBuyerStarted {
+		return nil, grpc.Errorf(codes.FailedPrecondition, "ticket buyer is already started")
+	}
+	return &pb.StartTicketPurchaseResponse{}, err
+}
+
+// StopTicketPurchase disables ticket purchases in the ticket buyer.
+func (t *ticketBuyerServer) StopTicketPurchase(ctx context.Context, req *pb.StopTicketPurchaseRequest) (
+	*pb.StopTicketPurchaseResponse, error) {
+
+	err := t.loader.StopTicketPurchase()
+	if err == loader.ErrTicketBuyerStopped {
+		return nil, grpc.Errorf(codes.FailedPrecondition, "ticket buyer is not running")
+	}
+	return &pb.StopTicketPurchaseResponse{}, err
+}
+
+// Config returns the configuration of the ticket buyer.
+func (t *ticketBuyerServer) Config(ctx context.Context, req *pb.TicketBuyerConfigRequest) (
+	*pb.TicketBuyerConfigResponse, error) {
+	pm := t.loader.PurchaseManager()
+	if pm == nil {
+		return nil, grpc.Errorf(codes.FailedPrecondition, "ticket buyer is not running")
+	}
+	config := pm.Purchaser().Config()
+	return &pb.TicketBuyerConfigResponse{
+		AccountName:           config.AccountName,
+		AvgPriceMode:          config.AvgPriceMode,
+		AvgPriceVWAPDelta:     int64(config.AvgPriceVWAPDelta),
+		BalanceToMaintain:     int64(config.BalanceToMaintainAbsolute),
+		BlocksToAvg:           int64(config.BlocksToAvg),
+		DontWaitForTickets:    config.DontWaitForTickets,
+		ExpiryDelta:           int64(config.ExpiryDelta),
+		FeeSource:             config.FeeSource,
+		FeeTargetScaling:      config.FeeTargetScaling,
+		MinFee:                int64(config.MinFee),
+		MinPriceScale:         config.MinPriceScale,
+		MaxFee:                int64(config.MaxFee),
+		MaxPerBlock:           int64(config.MaxPerBlock),
+		MaxPriceAbsolute:      int64(config.MaxPriceAbsolute),
+		MaxPriceRelative:      config.MaxPriceRelative,
+		MaxPriceScale:         config.MaxPriceScale,
+		MaxInMempool:          int64(config.MaxInMempool),
+		PoolAddress:           config.PoolAddress,
+		PoolFees:              config.PoolFees,
+		SpreadTicketPurchases: config.SpreadTicketPurchases,
+		TicketAddress:         config.TicketAddress,
+		TxFee:                 int64(config.TxFee),
+	}, nil
+}
+
+// SetAccountName sets the account name to use for purchasing tickets.
+func (t *ticketBuyerServer) SetAccountName(ctx context.Context, req *pb.SetAccountNameRequest) (
+	*pb.SetAccountNameResponse, error) {
+
+	pm := t.loader.PurchaseManager()
+	if pm == nil {
+		return nil, grpc.Errorf(codes.FailedPrecondition, "ticket buyer is not running")
```

Comment:
since this pattern is repeated so much please make a helper function to return a required purchase manager or error with a grpc error code